app/src/main/java/com/infomaniak/drive/ui/publicShare/OnPublicShareItemClickListener.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Infomaniak kDrive - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.drive.ui.publicShare
+
+import androidx.annotation.StringRes
+import com.infomaniak.drive.R
+import com.infomaniak.drive.data.models.File
+import com.infomaniak.drive.ui.fileList.BaseDownloadProgressDialog.DownloadAction
+import com.infomaniak.drive.ui.fileList.preview.PreviewDownloadProgressDialogArgs
+import com.infomaniak.drive.ui.fileList.preview.PreviewPDFHandler
+import com.infomaniak.drive.utils.DrivePermissions
+import com.infomaniak.drive.utils.IOFile
+import com.infomaniak.drive.views.FileInfoActionsView
+import com.infomaniak.drive.views.FileInfoActionsView.OnItemClickListener.Companion.downloadFile
+import com.infomaniak.lib.core.utils.safeNavigate
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+interface OnPublicShareItemClickListener : FileInfoActionsView.OnItemClickListener {
+
+    val drivePermissions: DrivePermissions
+    val publicShareViewModel: PublicShareViewModel
+    val previewPDFHandler: PreviewPDFHandler?
+
+    fun initCurrentFile()
+    fun onDownloadSuccess()
+    fun onDownloadError(@StringRes errorMessage: Int)
+
+    override fun openWith() {
+        executeActionAndClose(DownloadAction.OPEN_WITH)
+    }
```

Comment:
```suggestion
        override fun openWith() = executeActionAndClose(DownloadAction.OPEN_WITH)
```

You can do that on one line for other methods.